### PR TITLE
Improved tests performance

### PR DESCRIPTION
**Pull request by MaksymilianDemitraszek**

### CODE REVIEW

File: src/commands/test/contract_fork.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+class ForkableStarknet(Starknet):
+    @classmethod
+    async def empty(
+        cls, general_config: Optional[StarknetGeneralConfig] = None
+    ) -> "Starknet":
+        return ForkableStarknet(
+            state=await StarknetState.empty(general_config=general_config)
+        )
+
+    def plug_from_different_state(self, deployed_contact: StarknetContract):
```
Comment (kasperski95) [inline src/commands/test/contract_fork.py:19]:
```suggestion
    def copy_and_adapt_contract(self, starknet_contract: StarknetContract):
```
- [ ] fix typo: `contact` -> `contract`
- Does the contract have to be deployed?
- The name of this function is a bit confusing. I'd change it to `copy_and_adapt_contract`.

File: src/commands/test/contract_fork.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import copy
```
Comment (kasperski95) [inline src/commands/test/contract_fork.py:1]: Consider changing the name of this file to `forkable_starknet.py`.

File: src/commands/test/runner.py [hunk 3]
```diff
@@ -159,10 +158,24 @@ async def empty(
...
-        env.starknet = await Starknet.empty()
+        env.starknet = await ForkableStarknet.empty()
         env.test_contract = await env.starknet.deploy(contract_def=test_contract)
         return env
 
+    def fork(self):
+        assert self.starknet
+        assert self.test_contract
+
+        n_env = TestExecutionEnvironment(
```
Comment (kasperski95) [inline src/commands/test/runner.py:169]: What does the `n` mean in this case? `new`?

File: src/commands/test/runner.py [hunk 4]
```diff
@@ -159,10 +158,24 @@ async def empty(
...
+            is_test_fail_enabled=self._is_test_fail_enabled,
```
Comment (kasperski95) [inline src/commands/test/runner.py:170]: I think the `is_test_fail_enabled` is no longer used.

**Review by kasperski95 — APPROVED** [review]
LGTM

File: src/commands/test/contract_fork.py (hunk 1 above)
Comment (MaksymilianDemitraszek) [reply src/commands/test/contract_fork.py:19]: Yea, name is terrible, your is much better, thanks. Contact doesn't have to be deployed, we copy state

File: src/commands/test/runner.py (hunk 3 above)
Comment (MaksymilianDemitraszek) [reply src/commands/test/runner.py:169]: we can rename it to new

File: src/commands/test/runner.py (hunk 4 above)
Comment (MaksymilianDemitraszek) [reply src/commands/test/runner.py:170]: True, merge conflicts

**Pull request merged**